### DISCUSSION TITLE
how to remember hls history?

### DISCUSSION THREAD

**Discussion by Kamilcuk** [bluenviron/mediamtx#2060] [Questions] [answered]
Hi, I would want persistent hls history.

When a streamer connects, disconnects and connects again, the hls starts writing from the first chunk and website looses history.

I would want history to be persistent. What can I do? I tried hlsAlwaysRemux but I do not think it worked. Should it work with hlsAlwaysRemux option?

Thanks. And mediamtx is great, thanks for making it.

**Comment by aler9** [comment, marked as answer]
Hello, this is a missing feature requires another missing feature, that is the ability to natively save streams on disk, that is being tracked here: #1399

After we implement native recording, we can add to the server any feature that requires access to the time dimension, including this one. Currently we can only implement features that work with real-time streams.

In the meanwhile, you can record your streams in MPEG-TS or MP4 (https://github.com/bluenviron/mediamtx#save-streams-on-disk) and serve them with an HTTP server like Nginx.

**Comment by Kamilcuk** [comment]
Hi, amazing, thanks for your response.
So what I ended up doing, is to run ffmpeg to read from the stream with a lot of -hls options and to generate m3u8 file with ffmpeg. Funny enough, ffmpeg has all the features - like cleaning old partial recordings, resuming m3u8 file, etc. Then I setup up nginx with index.html with videojs to stream the m3u8 on my website.
Still, mediamtx is used on the receiver side, because of passwords and access and configuration, and it is very stable. Thanks!

[a comment by an automated account is not shown]
